### ISSUE TITLE
Or filtering does not respect the AND filters

### ISSUE THREAD

**Issue by alechirsch** [scoutforpets/bookshelf-jsonapi-params#65]
Given the below filter:
```
{
  active: true,
  or: [
   {
     account_id: 1
   },
   {
     group_id: 5
   }
  ]
}
```
Currently the generated SQL for the where clause will produce this: 
```
where
  active=true
  or ( account_id=1) or (group_id=5)
```

The generated SQL should instead look like this:
```
where
  active=true
  and ( ( account_id=1) or (group_id=5) ) 
```

Related to #63 

@massimo-ua I found this when trying to use the or filter feature, I will be looking into a fix for this. Can you confirm your intentions when building out the feature? I do not want to break anything for you. Everything outside of the `or` array should be `and`'ed together.

**Comment by massimo-ua** [comment]
@alechirsch Yes, you are completely right. Such kind of behavior (my impl) is not obvious and yours is looking much better and straightforward. Can I offer help in fixing this implementation?

**Comment by alechirsch** [comment]
Released as 1.5.1
